detection_rules/mappings.py

```diff
@@ -73,3 +77,79 @@ def get_rta_files(self, rta_list=None, rule_ids=None):
                 rta_files.add(rta_path)
 
         return list(sorted(rta_files))
+
+
+def get_triggered_rules() -> dict:
+    """Get the rules that are triggered by each rta."""
+    triggered_rules = {}
+    for rta_test in get_available_tests()[1]:
+        for rule_info in rta_test.get("siem"):
+            rule_id = rule_info.get("rule_id")
+            for platform in rta_test.get("platforms"):
+                triggered_rules.setdefault(platform, []).append(rule_id)
+    return triggered_rules
+
+
+def get_platform_list(rule: TOMLRule) -> list:
+    """Get the list of OSes for a rule."""
```

Comment:
Since the `os_list` in `metadata` is optional and rarely used, it may make sense to use tags (until we add os to all queries in endpoint specific rules)

Reply:
#2286

Reply:
I'm going to leave this as-is for now since it tries to do both based on available metadata.